### ISSUE TITLE
SQL query is printed into browser in case of exception

### ISSUE THREAD

**Issue by alexkuk** [magento/magento2#13385]
### Preconditions
1. Magento 2.2.2. Was reproduced on Magento 2.1.9 as well.
2. Production mode
3. Reproduced on EAV collections

### Steps to reproduce
1. Add an erroneous SQL statement to collection select object. I've added this code to \Magento\Catalog\Block\Product\ListProduct::initializeProductCollection():
```php
$collection = $layer->getProductCollection();
$collection->getSelect()->columns('qwerty');
```
as an example
2. Open some category page in a browser.

### Expected result
1. No SQL query is printed in a browser

### Actual result
1. SQL query is printed in a browser

### Fix proposal
In the \Magento\Eav\Model\Entity\Collection\AbstractCollection::_loadEntities() replace `$this->printLogQuery(true, true, $query);` with `$this->printLogQuery(false, true, $query);`

**Comment by bydrei** [comment]
@alexkuk do you have pub/errors/local.xml or pub/errors/local.xml.sample ?

**Comment by magento-engcom-team** [comment]
@alexkuk, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by shyamranpara** [comment]
I have printed following query
`$collection = $layer->getProductCollection();
 $collection->getSelect()->columns('qwerty');
 echo (string) $collection->getSelect();`

and I found following result in printed query on screen
 `e`.`qwerty` FROM `catalog_product_entity` AS `e` INNER JOIN `catalog_category_product_index` 

it tries to select column "qwerty" from this "catalog_product_entity" table, column is not exist in this table.
so it is normal behaviour of magento throws exception **Column not found: 1054 Unknown column 'e.qwerty' in 'field list'** and display in frontend in category page.

it is not recommended to change any code in production mode and test in production mode. 

This is not bug this is feature that display exception on frontside.

**Comment by alexkuk** [comment]
@shyamranpara Common, displaying SQL in production IS an issue. It is a security problem and it's acknowledged and reproduced by the core team. The problem is in the AbstractCollection that is used by both framework and application code. Changing the core file is provided as an example how to reproduce the bug.

**Comment by magento-engcom-team** [comment]
Hi @alexkuk. Thank you for your report. 
The issue has been fixed in magento/magento2#13607 by @shyamranpara in 2.2-develop branch
Related commit(s):
 - [f473572a997812b8d61ec0ee3f000a569857b263](https://github.com/magento/magento2/commit/f473572a997812b8d61ec0ee3f000a569857b263)

The fix will be available with the upcoming 2.2.5 release.

**Comment by magento-engcom-team** [comment]
Hi @alexkuk. Thank you for your report. 
The issue has been fixed in magento/magento2#14223 by @rostyslav-hymon in 2.3-develop branch
Related commit(s):
 - [f618e14d145f5f5a1751c0108c1f9e1b94f50ea6](https://github.com/magento/magento2/commit/f618e14d145f5f5a1751c0108c1f9e1b94f50ea6)

The fix will be available with the upcoming 2.3.0 release.
